express/express/route.cpp

```
@@ -0,0 +1,61 @@
+#include "route.h"
+#include <iostream>
+
+using namespace CExpress;
+
+Route::Route(RequestType t, std::string p) {
+    if (p.empty() || p.size() == 1 || p[0] != '/') {
+        throw "Input path invalid.";
+    }
+
+    type = t;
+    path = p.find_last_of('/') == p.size() - 1 ? p.substr(0, p.size() - 1) : p;
+}
+
+std::string Route::getPath() const { return path; }
+RequestType Route::getRequestType() const { return type; }
+
+bool Route::matches(const Route& target) const {
+    if (getRequestType() != target.getRequestType()) {
+        return false;
+    }
+    
+    std::string sourcePath = getPath();
+    std::string targetPath = target.getPath();
+        
+    std::vector<std::string> sourceTokens;    
+    std::vector<std::string> targetTokens;
+    
+    splitString(sourcePath, '/', targetTokens);	
+    splitString(targetPath, '/', sourceTokens);
+
+    if (sourceTokens.size() != targetTokens.size()) {
+        return false;
+    }
+    
+    for (int i = 0; i < (int)sourceTokens.size(); ++i) {
+        if ((sourceTokens[i].find_first_of(':') == 0) != (targetTokens[i].find_first_of(':') == 0)) {
+            continue;
+        }
+        if (sourceTokens[i] != targetTokens[i]) {
+            return false;
+        }
+    }
+    
+    return true;
+}
+
+template<typename Out>
+void Route::split(const std::string &s, const char delim, Out result) const {
+	std::stringstream ss;
+	ss.str(s);
```

doing so throws an exception when creating route with input "//" (which is invalid, but should throw the "invalid path" exception)